lib/msal-node/src/client/ConfidentialClientApplication.ts

```diff
@@ -97,6 +97,15 @@ export class ConfidentialClientApplication extends ClientApplication implements
             clientAssertion
         };
 
+        /*
+         * valid request should not have "common" or "organization" in lieu of the tenant_id in the authority in the auth configuration
+         * example authority: "https://login.microsoftonline.com/TenantId",
+         */
+        const tenantId = validRequest.authority.split("/")[3];
+        if ((tenantId === "common") || (tenantId === "organization")) {
```

Comment:
@Robbie-Microsoft @bgavrilMS I know it doesn't make sense with client credentials, but do we know what is the behavior for `consumers`. Should we check for this as well?

Reply:
Can we test it?

Reply:
Ok, looks like it doesn't give an error actually. It's the same behavior as if using common or organizations.

Reply:
Yes, I think we should -it's less likely for someone to use consumers here but if they do, it's hard to figure out what's wrong from the response. We should also use the AADAuthorityConstants here instead of hardcoding it.